src/evm/EvmDiamondStateMachine.ts

```diff
@@ -134,37 +134,29 @@ class EvmDiamondStateMachine extends ADiamondStateMachine {
     async getParticipants(): Promise<Address[]> {
         const callData = this.getEncodedCalldata("getParticipants");
 
-        let result = await this.contractExecuter.executeCall(callData);
-        const hexResult = hexlify(result.returnValue);
-        const [addresses] = ethers.AbiCoder.defaultAbiCoder().decode(
-            ["address[]"],
-            hexResult
+        const addresses = Codec.decodeEthersResult<Address[]>(
+            await this.contractExecuter.executeCall(callData),
```

Comment:
what happened to `hexResult = hexlify(result.returnValue)` do we not need to do it anymore? Would also rename `decodeEthersResult` -> `decodeEvmResult`

Reply:
>do we not need to do it anymore?

No, added a test to reassure you that it works

>rename decodeEthersResult -> decodeEvmResult

ok. done